helix-core/src/main/java/org/apache/helix/manager/zk/ZkBaseDataAccessor.java

```diff
@@ -85,13 +88,20 @@ public AccessResult() {
 
   private final HelixZkClient _zkClient;
 
+  //TODO: to be deprecated for client usage
   public ZkBaseDataAccessor(HelixZkClient zkClient) {
     if (zkClient == null) {
       throw new NullPointerException("zkclient is null");
     }
     _zkClient = zkClient;
   }
 
+  public ZkBaseDataAccessor(String zkAddress, ZkSerializer zkSerializer) {
+    _zkClient = new ZkClient.Builder()
+        .setConnection(new ZkConnection(zkAddress, ZkClient.DEFAULT_SESSION_TIMEOUT))
+        .setZkSerializer(zkSerializer).build();
```

Comment:
We talked about using SharedZkClientFactory here, right?

Reply:
Sure, I will check out the usage. It's one of the missing details with the current commit, I quickly draft it to simply show-case the new idea to see how the overall structure looks!